packages/demand-bloc-payment/src/PaymentBloc.ts

```diff
@@ -111,3 +132,24 @@ export class PaymentBloc {
     }
   }
 }
+
+const fetchPaymentProvider = async (paymentService: Payment) => {
+  const response = await paymentService.getPaymentProvider()
+  if (!response.ok) {
+    throw new Error('Cannot fetch payment provider')
+  }
+
+  return response.body
+}
+
+const getPaymentProvider = (
+  providers: PaymentProvidersMap,
+  paymentProviderResponse: PaymentProvidersResponse
+) => {
+  const targetId = paymentProviderResponse.provider?.id ?? ''
+  const found = Object.entries(providers).find(([id]) => id === targetId)
```

Comment:
can we do just `providers[targetId]` ?

Reply:
If we limit `providers` object with known IDs, we can't test this function against unknown ID since TS will complain about wrong key.